### DISCUSSION TITLE
next build command shows unhandledRejection Error: Cannot find module 'next/dist/shared/lib/html-context.js'

### DISCUSSION THREAD

**Discussion by rahuls-gfg** [vercel/next.js#47594] [Help]
### Summary

Hello, 

I have a project built using next 12.1. I'm trying to build the application using  ``` next build ```. The command shows the the following error: 

unhandledRejection Error: Cannot find module 'next/dist/shared/lib/html-context.js'
Require stack:
- /Users/rahul/projects/next-project/.next/server/pages/_document.js
- /opt/homebrew/lib/node_modules/next/dist/server/require.js
- /opt/homebrew/lib/node_modules/next/dist/server/load-components.js
- /opt/homebrew/lib/node_modules/next/dist/export/worker.js
- /opt/homebrew/lib/node_modules/next/dist/build/worker.js
- /opt/homebrew/lib/node_modules/next/dist/compiled/jest-worker/processChild.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1090:15)
    at mod._resolveFilename (/opt/homebrew/lib/node_modules/next/dist/build/webpack/require-hook.js:23:32)
    at Module._load (node:internal/modules/cjs/loader:934:27)
    at Module.require (node:internal/modules/cjs/loader:1157:19)
    at require (node:internal/modules/helpers:119:18)
    at 8743 (/Users/rahul/projects/next-project/.next/server/pages/_document.js:335:18)
    at __webpack_require__ (/Users/rahul/projects/next-project/.next/server/webpack-runtime.js:25:42)
    at 2327 (/Users/rahul/projects/next-project/.next/server/chunks/439.js:44:20)
    at __webpack_require__ (/Users/rahul/projects/next-project/.next/server/webpack-runtime.js:25:42)
    at 439 (/Users/rahul/projects/next-project/.next/server/chunks/439.js:1003:18) {
  type: 'Error',
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/rahul/projects/next-project/.next/server/pages/_document.js',
    '/opt/homebrew/lib/node_modules/next/dist/server/require.js',
    '/opt/homebrew/lib/node_modules/next/dist/server/load-components.js',
    '/opt/homebrew/lib/node_modules/next/dist/export/worker.js',
    '/opt/homebrew/lib/node_modules/next/dist/build/worker.js',
    '/opt/homebrew/lib/node_modules/next/dist/compiled/jest-worker/processChild.js'
  ]
}

I have tried running the next build command after deleting node_modules. Its still showing the same error. 

I have also attached file with detailed logs. [next-build.log](https://github.com/vercel/next.js/files/11088735/next-build.log)

Thank you.

**Comment by icyJoseph** [comment]
Is it normal the the require stack is pointing to `/opt/homebrew/lib/...`? Is it like, taking Next.js from your global scope?

I guess you have installed `next` globally, and calling `next build` directly from the command line causes issues?

What happens if you run `npm run build` instead? (Assuming you have next installed in your `/Users/rahul/projects/next-project` path).

**Reply by icyJoseph** [reply]
@rahuls-gfg awesome! please don't forget to mark as answered~~

**Comment by rahuls-gfg** [comment]
Hi @icyJoseph, you are right. It is taking from the global scope.

I have uninstalled next from global scope and ran ``` npm run build```, it is working now. 

Thank you.

**Comment by JClackett** [comment]
Im currently getting the same thing when deploying to vercel:

```
unhandledRejection Error: Cannot find module '../../next@13.2.4_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/shared/lib/constants.js'
--
11:29:52.436 | Require stack:
11:29:52.436 | - /vercel/path0/.next/server/pages/_document.js
11:29:52.436 | - /vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/require.js
11:29:52.436 | - /vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/server/load-components.js
11:29:52.436 | - /vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/worker.js
11:29:52.436 | - /vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/worker.js
11:29:52.436 | - /vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/jest-worker/processChild.js
11:29:52.436 | at Module._resolveFilename (node:internal/modules/cjs/loader:1039:15)
11:29:52.436 | at mod._resolveFilename (/vercel/path0/node_modules/.pnpm/next@13.2.4_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/build/webpack/require-hook.js:23:32)
```

latest next version and pnpm v8

**Reply by JClackett** [reply]
Triggering a redeploy seemed to work? random error :)

**Comment by jrgleason** [comment]
This also helped me when I built with `next build` it failed similarly. I removed my .next folder and rebuilt with `npm run build` and everything worked thank you! It might be a good idea to open a ticket for documentation.
